### PR TITLE
New handler for either include files or program files

### PR DESCRIPTION
**Pull request by RaoulHC**
Handler that takes advantage of camfort/fortran-src#227.

### CODE REVIEW

File: src/Language/Fortran/Extras.hs [hunk 1]
```diff
@@ -203,6 +215,24 @@ withToolOptionsAndProgramAnalysis programDescription programHeader toolOptsParse
...
+-- or 'Programfile A0', run the handler on the appropriately parsed source,
+-- parsing anything that has the ".inc" extension as an include
+withToolOptionsAndProgramOrBlock
+  :: String
+  -> String
+  -> Parser a
+  -> (a -> Either (FilePath, [Block A0]) (ProgramFile A0) -> IO ())
+  -> IO ()
+withToolOptionsAndProgramOrBlock desc header optsParser handler = do
+  (srcOptions, toolOptions) <-
```
Comment (raehik) [inline src/Language/Fortran/Extras.hs:229]: Nag: bit clearer as `RunOptions srcOptions toolOptions <- getRunOptions desc header optsParser` (or by binding then extracting manually)

**Review by raehik — CHANGES_REQUESTED** [review]

**Comment by raehik** [comment]
I tend to throw Git repo references into `stack.yaml` when trying to use code that isn't yet on Hackage. Like so:

```yaml
resolver: nightly-2022-03-30
packages: [.]

extra-deps:
# not yet released :<
- git: https://github.com/nikita-volkov/refined
  commit: 91566f4e313937b7315a401fd116fa0aa2a15f06
```

Adding a clause for fortran-src would fix the CI while a Hackage release is pending (there's similar syntax for Cabal). I can add that to this PR or after merge.

File: src/Language/Fortran/Extras.hs (hunk 1 above)
Comment (raehik) [reply src/Language/Fortran/Extras.hs:229]: I learned how to do suggested changes lol

File: src/Language/Fortran/Extras.hs [hunk 2]
```diff
@@ -203,6 +215,24 @@ withToolOptionsAndProgramAnalysis programDescription programHeader toolOptsParse
...
+    (fortranSrcOpts &&& toolOpts) <$> getRunOptions desc header optsParser
```
Comment (raehik) [inline src/Language/Fortran/Extras.hs:230]: Nag: bit clearer with pattern matching (or by binding then extracting manually). Arrow import probably unused with this.

```suggestion
  RunOptions srcOptions toolOptions <- getRunOptions desc header optsParser
```

Comment (RaoulHC) [reply src/Language/Fortran/Extras.hs]: Changed to this, probably should have seen this rather than reaching for arrows 😅

**Comment by raehik** [comment]
Cheers, I'll merge this!

**Comment by RaoulHC** [comment]
Looks like the cabal test was still failing, does something else need to be added for that?

**Comment by raehik** [comment]
yeah, easiest to merge and copy over config from another project - done in latest commit https://github.com/camfort/fortran-src-extras/commit/7caee0797b35bcc65d0fa7f74c77e797567a573c

**Pull request merged**